### PR TITLE
completed tasks 1-2

### PR DESCRIPTION
**Pull request by laurieclif**
task 2 - created and tested /api/topics endpoint

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+ describe("/api/topics", () => {
+    test("GET 200: responds with an array of topic objects, each with slug and description properties", () => {
+        return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({body}) => {
+            expect(body.length).toBe(3)
+            body.forEach((topic) => {
+                expect(topic).toHaveProperty("description")
```
Comment (roisinoreilly) [inline __tests__/app.test.js:24]: a better jest matcher for testing objects, especially as you'll start testing bigger ones in the next few tickets, is .toMatchObject or .objectContaining. I'd suggest having a go at using them on future tickets

File: app.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const express = require('express')
+const app = express()
+const { getTopics } = require('./controllers/controllers.js')
+
+app.get("/api/topics", getTopics)
+
+app.all('*', (req, res, next) => {
```
Comment (roisinoreilly) [inline app.js:7]: perfect

File: controllers/controllers.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const { fetchTopics } = require("../models/models.js");
+
+const getTopics = (req, res, next) => {
+    return fetchTopics().then((topics) => {
+        res.status(200).send(topics)
```
Comment (roisinoreilly) [inline controllers/controllers.js:5]: anything from a controller needs to be sent back on a meaningful key ie. .send({topics})

**Review by roisinoreilly — APPROVED** [review]
great work on this ticket and well done with the 404 handler. I've left a few comments to improve your tests but otherwise keep it up!

File: __tests__/app.test.js (hunk 1 above)
Comment (laurieclif) [reply __tests__/app.test.js:24]: thanks for the comments Roisin, I did try toMatchObject but it wasn't working the way I expected it to so resorted to using toHaveProperty but I can try it again

**Pull request merged**